src/player.js

```diff
@@ -486,12 +502,42 @@ export default class Player extends FakeEventTarget {
     }
   }
 
+  /**
+   * Resets the necessary components before change media.
+   * @private
+   * @returns {void}
+   */
+  reset(): void {
+    if (this._reset) return;
+    this.dispatchEvent(new FakeEvent(CustomEventType.RESET_STARTED));
+    this.pause();
+    this._eventManager.removeAll();
+    this._createReadyPromise();
+    this._activeTextCues = [];
+    this._updateTextDisplay([]);
+    this._tracks = [];
```

Comment:
I think we can move the state attributes to an internal method and call them from reset and destroy as all of them appear in both methods.

Reply:
done